File: CS/WindowsFormsApplication32/MyTabbedMdiManager/MySkinTabHeaderViewInfo.cs
using System.Drawing;
using DevExpress.XtraTab;
using DevExpress.XtraTab.Buttons;
using DevExpress.XtraTab.ViewInfo;

namespace CustomTabbedMdiManager
{
	public class MySkinTabHeaderViewInfo : SkinTabHeaderViewInfo
	{
		public MySkinTabHeaderViewInfo(BaseTabControlViewInfo viewInfo)
			: base(viewInfo)
		{
		}

		protected override TabButtonsPanel CreateHeaderButtons()
		{
			return new MyTabButtonsPanel(ViewInfo);
		}

		protected override Size CalcButtonsSize(TabButtons btns)
		{
			Size buttonsSize = base.CalcButtonsSize(btns);
			if ( buttonsSize == Size.Empty && ViewInfo.AddActiveFilesButton )
				buttonsSize = Buttons.CalcSize(GraphicsInfo.Graphics);

			return buttonsSize;
		}

		public new MySkinTabControlViewInfo ViewInfo
		{
			get
			{
				return base.ViewInfo as MySkinTabControlViewInfo;
			}
		}
	}
}

File: VB/WindowsFormsApplication32/MyTabbedMdiManager/MySkinTabHeaderViewInfo.vb
Imports System.Drawing
Imports DevExpress.XtraTab
Imports DevExpress.XtraTab.Buttons
Imports DevExpress.XtraTab.ViewInfo

Namespace CustomTabbedMdiManager

    Public Class MySkinTabHeaderViewInfo
        Inherits SkinTabHeaderViewInfo

        Public Sub New(ByVal viewInfo As BaseTabControlViewInfo)
            MyBase.New(viewInfo)
        End Sub

        Protected Overrides Function CreateHeaderButtons() As TabButtonsPanel
            Return New MyTabButtonsPanel(ViewInfo)
        End Function

        Protected Overrides Function CalcButtonsSize(ByVal btns As TabButtons) As Size
            Dim buttonsSize As Size = MyBase.CalcButtonsSize(btns)
            If buttonsSize = Size.Empty AndAlso ViewInfo.AddActiveFilesButton Then buttonsSize = Buttons.CalcSize(GraphicsInfo.Graphics)
            Return buttonsSize
        End Function

        Public Overloads ReadOnly Property ViewInfo As MySkinTabControlViewInfo
            Get
                Return TryCast(MyBase.ViewInfo, MySkinTabControlViewInfo)
            End Get
        End Property
    End Class
End Namespace
